src/components/ProductGrid/ProductGrid.js

```diff
@@ -1,12 +1,13 @@
 import React, { Component } from "react";
 import PropTypes from "prop-types";
 import Grid from "material-ui/Grid";
-
 import ProductItem from "components/ProductItem";
+import PageStepper from "components/PageStepper";
 
-class ProductGrid extends Component {
+export default class ProductGrid extends Component {
   static propTypes = {
-    catalogItems: PropTypes.arrayOf(PropTypes.object)
+    catalogItems: PropTypes.arrayOf(PropTypes.object),
```

Comment:
Making a prop required doesn't stop it the component from being rendered.

If I make this required, and `catalogItems` is undefined, which it can be, there will be a warning in the component, and potentially crashing the app if I also remove the conditional the blocks the prevent that rendering. 

I'll still have to put a conditional somewhere to handle it. This eventually leads to needing a `Products not found` view. Should this component render that `not-found` view if it doesn't have any available? Or is that to be handled by another component.

Reply:
Perhaps we could have a no found view, @aldeed what are your thoughts an this issue? This kind of situation could need a pattern, as it occurs often when rendering lists.